### PR TITLE
[SPARK-44780][DOC] SQL temporary variables

### PR DESCRIPTION
**Pull request by srielau**
### What changes were proposed in this pull request?

Document the previously pushed feature SQL temporary variables

### Why are the changes needed?

If it's not documented, it doesn't exist....

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

build docs, verify HTML

### CODE REVIEW

**Comment by srielau** [comment]
Screenshot 2023-08-11 at 10 26 54 PM
Screenshot 2023-08-11 at 10 24 33 PM
Screenshot 2023-08-11 at 10 22 55 PM

**Comment by srielau** [comment]
@cloud-fan I think this is ready.

**Review by yaooqinn — APPROVED** [review]

File: docs/sql-ref-syntax-ddl-declare-variable.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+### Parameters
+
+* **OR REPLACE**
+
+    If specified, a pre-existing temporary variable is replaced if it exists.
+
+* **variable_name**
+
+    Specifies a name for the variable to be created.
+    The function name may be optionally qualified with a `system`.`session` or `session`.
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-ddl-declare-variable.md:50]:
```suggestion
    The variable name may be optionally qualified with a `system`.`session` or `session`.
```

File: docs/sql-ref-syntax-ddl-declare-variable.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+* **default_expr**
+
+    An optional expression used to initialize the value of the variable after declaration.
+    The expression is re-evaluated whenever the variable is reset to `DEFAULT` using
+    [SET VAR](sql-ref-syntax-aux-set-var.html).
+    If `data_type` is specified `default_expr` must be castable to the variable type.
+    If `data_type` is not specified you must specify a default and its type will become the type of
+    the variable.
+    If no default expression is given, the variable is initialized with `NULKL`.
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-ddl-declare-variable.md:67]:
```suggestion
    If no default expression is given, the variable is initialized with `NULL`.
```

**Review by cloud-fan — APPROVED** [review]
LGTM except for two typos.

File: docs/sql-ref-syntax-aux-set-var.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+The `SET VAR` command sets a temporary variable which has been previously declared in the current session.
+
+To set a config variable or a hive variable use [SET](sql-ref-syntax-aux-conf-mgmt-set.html).
+
+### Syntax
+
+```sql
+SET { VAR | VARIABLE }
+  { variable_name = { expression | DEFAULT } |
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-aux-set-var.md:32]: After looking at the parser rule, I think we support `SET VAR a=1, b=2, ...`

Comment (cloud-fan) [reply docs/sql-ref-syntax-aux-set-var.md:32]: do we want to change the parser and forbid it?

**Comment by HyukjinKwon** [comment]
👍

File: docs/sql-ref-syntax-aux-set-var.md (hunk 3 above)
Comment (srielau) [reply docs/sql-ref-syntax-aux-set-var.md:32]: *gosh* I forgot.

Comment (srielau) [inline docs/sql-ref-syntax-aux-set-var.md:32]:
```suggestion
  { variable_name = { expression | DEFAULT } [, ...] |
```

File: docs/sql-ref-syntax-aux-set-var.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+The `SET VAR` command sets a temporary variable which has been previously declared in the current session.
+
+To set a config variable or a hive variable use [SET](sql-ref-syntax-aux-conf-mgmt-set.html).
+
+### Syntax
+
+```sql
+SET { VAR | VARIABLE }
+  { variable_name = { expression | DEFAULT } [, ...] |
```
Comment (srielau) [inline docs/sql-ref-syntax-aux-set-var.md:32]:
```suggestion
  { { variable_name = { expression | DEFAULT } } [, ...] |
```

**Comment by srielau** [comment]
OK, let's merge @cloud-fan

File: docs/sql-ref-syntax-ddl-drop-variable.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `DROP TEMPORARY` statement drops a temporary variable. An exception will
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-ddl-drop-variable.md:24]:
```suggestion
The `DROP TEMPORARY VARIABLE` statement drops a temporary variable. An exception will
```

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master! (doc only, no need to test)

**Pull request closed without merging**